### ISSUE TITLE
File("whatever.pvd", restart=1) doesn't work in parallel

### ISSUE THREAD

**Issue by pefarrell** [firedrakeproject/firedrake#1366]
Consider the code

```
from firedrake import *

mesh = UnitSquareMesh(16, 16)
V = FunctionSpace(mesh, "CG", 1)

a = interpolate(Constant(0), V); a.rename("Solution")
b = interpolate(Constant(1), V); b.rename("Solution")

pvd = File("output.pvd")
pvd.write(a)
del pvd
import gc
gc.collect()

pvd = File("output.pvd", restart=1)
pvd.write(b)
```

In serial this works fine; `output.pvd` has two functions, one all 0 and one all 1. But in parallel, the PVD file is broken. When I open paraview, the first solution it shows is a mix of 0 and 1.

![brokenrestart](https://user-images.githubusercontent.com/3494657/51931425-a7758500-23f4-11e9-8053-ee11a65227ab.png)

I think this is because the calls to `next(self.counter)` and `next(self.timestep)` in `File.__init__` happen within a `elif self.comm.rank == 0` block.

**Comment by pefarrell** [comment]
This bug is superseded by https://github.com/firedrakeproject/firedrake/pull/1367 .

**Comment by wence-** [comment]
Closed since we removed this functionality in #1367.
